src/types/builder.d.ts

```diff
@@ -0,0 +1,58 @@
+import { Field, Operation, Display, Constant, ConfigArrayName } from './checker'
+
+export type BuilderActionType = 'ADD' | 'REMOVE' | 'UPDATE' | 'REORDER'
+
+export type BuilderAddPayload =
+  | FieldPayload
+  | OperationPayload
+  | DisplayPayload
+  | ConstantPayload
+
+export type BuilderUpdatePayload = (
+  | FieldPayload
+  | OperationPayload
+  | DisplayPayload
+  | ConstantPayload
+) & {
+  currIndex: number
+}
+
+export interface FieldPayload {
+  configArrName: 'fields'
+  element: Field
+}
+
+export interface OperationPayload {
+  configArrName: 'operations'
+  element: Operation
+}
+
+export interface DisplayPayload {
+  configArrName: 'displays'
+  element: Display
+}
+
+export interface ConstantPayload {
+  configArrName: 'constants'
+  element: Constant
+}
+
+export interface BuilderRemovePayload {
+  currIndex: number
+  configArrName: ConfigArrayName
```

Comment:
Given that you are assigning the ConfigArrayEnum, couldn't you just type `configArrName` to be ConfigArrayName?

Reply:
See comment above about jest and enums